### ISSUE TITLE
Preload link should include the "crossorigin" attribute, if original request uses CORS

### ISSUE THREAD

**Issue by kgilden** [dunglas/vulcain#82]
I've been trying to get Vulcain working in a situation where CORS is necessary. At the moment it seems to me that browsers are blocking preload requests, because the generated link headers do not include the `crossorigin` attribute ([docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content#cors-enabled_fetches)).

![2021-04-25_17-03](https://user-images.githubusercontent.com/918599/115996514-40050a80-a5e8-11eb-8481-e49fddf077b1.png)

I'd propose adding `crossorigin=anonymous`, if the `Access-Control-Allow-Origin` header is present. And `crossorigin=use-credentials`, if the `Access-Control-Allow-Credentials` header is `true` ([docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-crossorigin)).

However, I'm not entirely sure what should happen, if the related resources are owned by different API-s (e.g. `bookapi.com/book/1` is referencing `authorapi.com/author/2`). In that case the abovementioned could not be implied from the response headers. 

Since the `Link` headers are added by Vulcain, there's no way for the upstream to handle this or is there?

**Comment by mabezdek** [comment]
Some problem, some solution?

**Comment by dunglas** [comment]
I don't think it will be possible to support cross-domain references. However, PR welcome add a config option to add crossorigin attributes.
